alpha/lib/model/kuserPeer.php

```diff
@@ -637,6 +638,55 @@ public static function sendNewUserMail(kuser $user, $existingUser)
 			$bodyParams
 		);
 	}
+
+	public static function getUserMailType($authType, $existingUser)
+	{
+		$existingUserMailMap = array(KalturaPartnerAuthenticationType::SSO => self::KALTURA_NEW_EXISTING_USER_EMAIL,
+			KalturaPartnerAuthenticationType::TWO_FACTOR_AUTH => self::KALTURA_NEW_EXISTING_USER_2FA_EMAIL,
+			KalturaPartnerAuthenticationType::PASSWORD_ONLY => self::KALTURA_NEW_EXISTING_USER_EMAIL);
+
+		$newUserMailMap = array(KalturaPartnerAuthenticationType::SSO => self::KALTURA_NEW_USER_EMAIL,
+			KalturaPartnerAuthenticationType::TWO_FACTOR_AUTH => self::KALTURA_NEW_USER_2FA_EMAIL,
+			KalturaPartnerAuthenticationType::PASSWORD_ONLY => self::KALTURA_NEW_USER_EMAIL);
+
+		if($existingUser)
+		{
+			return $existingUserMailMap[$authType];
+		}
+		else
+		{
+			return $newUserMailMap[$authType];
+		}
+	}
+
+	public static function getUserBodyParams($authType, $existingUser, $userName, $creatorUserName, $publisherName, $loginEmail, $resetPasswordLink, $partnerId, $roleName, $puserId, $kmcLink, $contactLink, $beginnersGuideLink, $quickStartGuideLink)
+	{
+		switch($authType)
+		{
+			case KalturaPartnerAuthenticationType::PASSWORD_ONLY:
+			case KalturaPartnerAuthenticationType::SSO:
+			if($existingUser)
```

Comment:
Need to consolidate = 

prefix - $userName, $creatorUserName, $publisherName, $loginEmail,
Suffix - $partnerId, $publisherName, $publisherName, $roleName, $publisherName, $puserId, $kmcLink, $contactLink, $beginnersGuideLink, $quickStartGuideLink

Only difference is the case that get the resetPasswordLink